st2actions/st2actions/runners/remote_script_runner.py

```diff
@@ -78,3 +84,116 @@ def _get_remote_action(self, action_parameters):
                                         sudo=self._sudo,
                                         timeout=self._timeout,
                                         cwd=self._cwd)
+
+
+class ParamikoRemoteScriptRunner(BaseParallelSSHRunner):
+    def run(self, action_parameters):
+        remote_action = self._get_remote_action(action_parameters)
+
+        LOG.debug('Will execute remote_action : %s.', str(remote_action))
+        result = self._run(remote_action)
+        LOG.debug('Executed remote_action: %s. Result is : %s.', remote_action, result)
+        status = self._get_result_status(result, cfg.CONF.ssh_runner.allow_partial_failure)
+
+        return (status, result, None)
+
+    def _run(self, remote_action):
+        try:
+            copy_results = self._copy_artifacts(remote_action)
+        except:
+            # If for whatever reason there is a top level exception,
+            # we just bail here.
+            error = 'Failed copying content to remote boxes.'
+            LOG.exception(error)
+            _, ex, tb = sys.exc_info()
+            copy_results = self._generate_error_results(' '.join([error, str(ex)]), tb)
+            return copy_results
+        else:
```

Comment:
We could avoid else and one level of indentation since we already return in except.